components/Comments/ReportPost.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useRef, useState,useEffect } from 'react';
+import Flag from '../../icons/flag.svg'
+import { XIcon } from "@heroicons/react/outline";
+import toast from "react-hot-toast";
+import { signIn, useSession } from 'next-auth/react';
+import { createReportEmailTemplate } from '../../utils/createReportEmailTemplate';
+
+interface Props {
+    name:string;
+    body:string;
+    id:number;
+    email:string | null;
+    slug:string
+}
+
+export const ReportPost = (props:Props) => {
+
+  const { data:session } = useSession(); 
+  const { name, body, id, email, slug } = props;
+  const [isModalOpen, setModalOpen] = useState<boolean>(false);
+  const [comment, setComment] = useState('')
+
+  const handleOpenModal = () => setModalOpen(true);
+  const handleCloseModal = () => setModalOpen(false);   
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDialogElement>) => {
+    if (event.key === "Escape") {
+        handleCloseModal();
+    }
+  };
+
+  const handleSubmit = async (e:React.FormEvent) =>{
+    e.preventDefault();
```

Comment:
can we validate session here before running the logic? Will a user who is not logged in be able to send emails?

Reply:
Agreed! Will stop spam abuse hopefully.

Reply:
Hi. I validate the session when the user clicks the flag. I thought that would do the job,  but I can add it here too.